src/lib/Simulator.ts

```diff
@@ -0,0 +1,221 @@
+import { CapabilityImprovements, ContainmentPolicy } from "./PlayerActions"
+import { RandomEvent } from "./RandomEvents"
+import { NexTurnState, PlayerActions, SimulatorState, WorldSetup, WorldState } from "./SimulatorState"
+import { FakeNegativeBinomial } from "./Probabilities"
+
+export class Simulator {
+
+    private initialState: WorldSetup
+    private scaleFactor: number
+    private daysPerTurn: number
+    private currentState: WorldState
+    private history: WorldState[]
+
+    constructor(initialState: WorldSetup) {
+        this.initialState = initialState
+        this.daysPerTurn = 10
+        this.scaleFactor = initialState.gdpPerDay * 0.2
+        this.currentState = this.computeInitialWorldState()
+        this.history = []
+    }
+
+    /**
+     * Processes
+     */
+    nextTurn(actionsInTurn: PlayerActions): NexTurnState {
+
+        // TODO: add guards to guarantee that player actions and capability improvements match the expectations.
+        // Assumption: The frontend returns ALL the player actions that are enabled and that should be factored in.
+
+        // Create a new copy of the current state to avoid side effects that can pollute the history
+        const nextStateCandidate = this.clone(this.currentState)
```

Comment:
The ideia here is that we create a fresh copy of the current state (Sn), the candidate state, and in order to compute the next state (Sn+1) we do the following:
* apply the recurring effects of the active player actions to the candidate state;
* apply the one-time effects of the new player actions (i.e. player actions applied on this turn) to the candidate state;
* apply the one-time effects of any newly selected random events to the candidate state - should random events have a recurring effect?
* compute the updated number of infected, dead, and costs via the `computeNextWorldState` method;
* "commit" the candidate state, adding the state Sn to the history and "promoting" the candidate state to Sn+1;